File: Longest_Common_Prefix/roman_to_integer.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        prefix = str("")
        prev = str("")
        if len(strs)==1:
            return prefix 
        for s in strs[0]:
            prefix = prefix + s
            for i in range(1,len(strs)):
                for ind, c in enumerate(prefix):
                    if strs[i][ind]!=c:
                        return prev
            prev = prefix
        return prefix
    
class Solution_Others:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        prefix = strs[0]
        for string in strs[1:]:
            while string.find(prefix) != 0:
                prefix = prefix[:-1]
                if not prefix:
                    return ""
        return prefix

# Example usage:
if __name__ == "__main__":
    s = Solution_Others()
    strs=["ab","a"]
    print(s.longestCommonPrefix(strs))

File: Longest_Common_Prefix/roman_to_integer.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
using namespace std;

class Solution{
public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
        if (strs.empty()) {
            return "";
        }
        std::string prefix = strs[0];
        for (size_t i = 1; i < strs.size(); ++i) {
            while (strs[i].find(prefix) != 0) {
                prefix = prefix.substr(0, prefix.size() - 1);
                if (prefix.empty()) {
                    return "";
                }
            }
        }
        return prefix;
    }
};

class Solution_Others {
    public:
        string longestCommonPrefix(vector<string>& strs) {
            int n = strs.size();
            string res = strs[0];
    
            for (int i = 1; i < n; i++) {
                string str = "";
                for (int j = 0; j < min(res.length(), strs[i].length()); j++) {
                    if (strs[i][j] == res[j])
                        str += res[j];
                    else
                        break;
                }
    
                res = str;
            }
    
            return res;
        }
    };

// Example usage
int main() {
    Solution_Others solution;
    std::vector<std::string> strs = {"ab", "a"};
    std::cout << solution.longestCommonPrefix(strs) << std::endl;
    return 0;
}